src/main/kotlin/com/learnspigot/bot/command/StartvoteCommand.kt

```diff
@@ -0,0 +1,123 @@
+                                    ) && user.id != bot.selfUser.id
+                                ) {
+                                    usersReacted.add(user.id)
+                                    usersReaction.add(emo)
+                                }
+                            }
+                        } else {
+                            for (emo in projectEmojis.indices.reversed()) {
+                                for (user in bot.getTextChannelById(System.getenv("VOTE_CHANNEL_ID"))!!
+                                    .retrieveMessageById(message!!.id).complete()
+                                    .retrieveReactionUsers(Emoji.fromUnicode(projectEmojis[emo]))) if (!usersReacted.contains(
+                                        user.id
+                                    ) && user.id != bot.selfUser.id
+                                ) {
+                                    usersReacted.add(user.id)
+                                    usersReaction.add(emo)
+                                }
+                            }
+                        }
+                        usersReaction.forEach { num -> sum += num }
+                        val reputation = sum / usersReacted.size
+                        val profile: UserProfile = datastore.findUserProfile(it.user.id)
+                        profile.addRep(
+                            leaderboardManager, it.guild!!, type, reputation
+                        )
+                        datastore.save(profile)
+                        it.messageChannel.sendMessage("The vote has ended, <@${it.user.id}> received $reputation reputation${if (reputation > 1) "s" else ""}")
+                            .queue()
+                        val tags = mutableListOf<ForumTag>()
+                        bot.getThreadChannelById(it.channel!!.id)!!.appliedTags.forEach { t -> tags.add(t) }
+                        tags.add(bot.getForumChannelById(System.getenv("FOR_REVIEW_CHANNEL_ID"))!!.availableTags.find { t -> t.name == "Approved" }!!)
+                        bot.getThreadChannelById(it.channel!!.id)!!.manager.setAppliedTags(tags).queue()
+                        message!!.editMessageEmbeds(
+                            voteEmbed(type, it, true)
+                        ).queue()
+                    }
+                } else {
+                    it.reply("Please be a support team member and use it in the correct channel").queue()
+                }
+
+            }
+        }.queue()
+    }
+
+    fun voteEmbed(type: String, interaction: GenericCommandInteractionEvent, ended: Boolean): MessageEmbed {
+        return Embed {
+            title = "**VOTE FOR REPUTATION**"
+            description =
+                "<@${interaction.user.id}> has just created a ${if (type == "Tutorial") "tutorial" else "project"}, please vote what reputation to give them between ${if (type == "Tutorial") "0-5" else "0-10"} based on quality and other aspects you find important. \n\n [Link](${
```

Comment:
I think we should abstract type perhaps into an enum. This can contain the name, the points range, etc.

Reply:
it is not gonna be many types as for now?

Reply:
as for now... we want this to be expandable, it is quite possible that we introduce more types. It also makes the code more readbale

Reply:
so i just make a enum with range, name, and put it in the parameter, is that ok